x-pack/plugins/ml/server/saved_objects/sync_task.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+  TaskInstance,
+} from '../../../task_manager/server';
+import type { SecurityPluginSetup } from '../../../security/server';
+import { savedObjectClientsFactory } from './util';
+import { jobSavedObjectServiceFactory } from './service';
+import { syncSavedObjectsFactory } from './sync';
+
+const SAVED_OBJECTS_SYNC_TASK_TYPE = 'ML:saved-objects-sync';
+const SAVED_OBJECTS_SYNC_TASK_ID = 'ML:saved-objects-sync-task';
+const SAVED_OBJECTS_SYNC_INTERVAL_DEFAULT = '1h';
+
+export class SavedObjectsSyncService {
+  private core: CoreStart | null = null;
+  private log: { [l: string]: (t: string, e?: Error) => void };
+
+  constructor(logger: Logger) {
+    this.log = createLocalLogger(logger, `Task ${SAVED_OBJECTS_SYNC_TASK_ID}: `);
+  }
+
+  public registerSyncTask(
+    taskManager: TaskManagerSetupContract,
+    security: SecurityPluginSetup | undefined,
+    spacesEnabled: boolean,
+    isMlReady: () => Promise<void>
+  ) {
+    taskManager.registerTaskDefinitions({
+      [SAVED_OBJECTS_SYNC_TASK_TYPE]: {
+        title: 'ML saved objet sync',
+        description: "This task periodically syncs ML's saved objects",
+        timeout: '1m',
+        maxAttempts: 3,
+        maxConcurrency: 1,
+
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              await isMlReady();
+              const core = this.core;
+              const { state } = taskInstance;
+
+              if (core === null || security === null || spacesEnabled === null) {
+                const error = 'dependencies not initialized';
+                this.log.error(error);
+                throw new Error(error);
+              }
+              const client = core.elasticsearch.client as unknown as IScopedClusterClient;
+
+              const { getInternalSavedObjectsClient } = savedObjectClientsFactory(
+                () => core.savedObjects
+              );
+              const savedObjectsClient = getInternalSavedObjectsClient();
+              if (savedObjectsClient === null) {
+                const error = 'Internal saved object client not initialized';
+                this.log.error(error);
+                throw new Error(error);
+              }
+
+              const jobSavedObjectService = jobSavedObjectServiceFactory(
+                savedObjectsClient,
+                savedObjectsClient,
+                spacesEnabled,
+                security?.authz,
+                client,
+                isMlReady
+              );
+              const { initSavedObjects } = syncSavedObjectsFactory(client, jobSavedObjectService);
+              const { jobs, trainedModels } = await initSavedObjects(false);
+              const count = jobs.length + trainedModels.length;
+
+              this.log.info(
+                count
+                  ? `${count} ML saved objects synced`
```

Comment:
Nit, can `objects` be output as `object` if `count === 1`, rather than `Task ML:saved-objects-sync-task: 1 ML saved objects synced`